eZ/Bundle/EzPublishCoreBundle/Command/ResizeOriginalImagesCommand.php

```diff
@@ -0,0 +1,136 @@
+        $ioService = $this->getContainer()->get('ezpublish.fieldType.ezimage.io_service.published');
+        $extensionGuesser = $this->getContainer()->get('liip_imagine.extension_guesser');
+
+        $contentType = $contentTypeService->loadContentTypeByIdentifier($contentTypeIdentifier);
+        $query = new Query();
+        $query->filter = new Query\Criterion\ContentTypeIdentifier($contentType->identifier);
+
+        $totalCount = $repository->sudo(function () use ($query, $searchService) {
+            return $searchService->findContent($query)->totalCount;
+        });
+
+        $output->writeln(sprintf('Found %d images matching given criteria.', $totalCount));
+
+        $progressBar = new ProgressBar($output, $totalCount);
+        $progressBar->start();
+
+        while ($query->offset <= $totalCount) {
+            $query->limit = $iterationCount;
+            $results = $repository->sudo(function () use ($query, $searchService) {
+                return $searchService->findContent($query);
+            });
+            $query->offset += $iterationCount;
+
+            /** @var SearchHit $hit */
+            foreach ($results->searchHits as $hit) {
+                try {
+                    /** @var Value $field */
+                    if (empty($hit->valueObject->fields[$imageField])) {
+                        $output->writeln(sprintf("<error>ContentType '%s' does not have '%s' field</error>",
+                            $contentType->identifier, $imageField));
+
+                        return;
+                    }
+                    foreach ($hit->valueObject->fields[$imageField] as $language => $field) {
+                        $binaryFile = $ioService->loadBinaryFile($field->id);
+                        $mimeType = $ioService->getMimeType($field->id);
+                        $binary = new Binary(
+                            $ioService->getFileContents($binaryFile),
+                            $mimeType,
+                            $extensionGuesser->guess($mimeType)
+                        );
+
+                        $recreated = $filterManager->applyFilter($binary, $variation);
+                        $this->store($ioService, $recreated, $field);
+                    }
+                } catch (NonExistingFilterException $e) {
+                    $output->writeln(sprintf('<error>%s</error>', $e->getMessage()));
+
+                    return;
+                } catch (\Exception $e) {
+                    $output->writeln(sprintf('<error>Can not resized image ID: %s, error message: %s</error>',
+                        $field->imageId, $e->getMessage()));
+                }
+
+                $progressBar->advance();
+            }
+        }
+
+        $progressBar->finish();
+        $output->writeln('');
+        $output->writeln(sprintf("<info>All images has been successfully resized using '%s' variation.</info>",
+            $variation));
```

Comment:
CS: this stands out on a diff, so maybe should be splitted into 4 lines (or more, I'd rely here on what PhpStorms would choose ;) ).